services/QuillLMS/app/workers/admin_diagnostic_reports/send_subscriptions_worker.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module AdminDiagnosticReports
+  class SendSubscriptionsWorker
+    include Sidekiq::Worker
+
+    def perform
+      subscriptions.find_each do |subscription|
+        SendSubscriptionCsvEmailWorker.perform_async(
+          subscription.user_id,
+          AdminReportFilterSelection::DIAGNOSTIC_GROWTH_SUBSCRIPTION_SHARED,
+          AdminReportFilterSelection::DIAGNOSTIC_GROWTH_SUBSCRIPTION_OVERVIEW,
+          AdminReportFilterSelection::DIAGNOSTIC_GROWTH_SUBSCRIPTION_SKILL,
+          AdminReportFilterSelection::DIAGNOSTIC_GROWTH_SUBSCRIPTION_STUDENT
+        )
+      end
+    end
+
+    private def subscriptions = (raise NotImplementedError)
```

Comment:
Oh cool!  I didn't know the syntax for raising exceptions with endless block

Reply:
I think you may be able to get away with not including the parentheses, too, but I feel like it reads a little easier this way.